test/unit/puppetlabs/puppetserver/lockable_pool_test.clj

```diff
@@ -508,17 +508,19 @@
           borrow-1 (.borrowItem pool)
           borrow-thread-started-2 (promise)
           borrow-thread-started-3? (promise)
+          lock-thread-locked? (promise)
           borrow-thread-2 (future
                            (deliver borrow-thread-started-2
                                     (Thread/currentThread))
+                           (timed-deref lock-thread-locked?)
                            (.borrowItem pool))
           borrow-thread-3 (future
                            (deliver borrow-thread-started-3? true)
+                           (timed-deref lock-thread-locked?)
                            (.borrowItem pool))
           borrow-thread-obj-2 @borrow-thread-started-2
           _ @borrow-thread-started-3?
           lock-thread-started? (promise)
-          lock-thread-locked? (promise)
           unlock? (promise)
           lock-thread (future
                        (deliver lock-thread-started? true)
```

Comment:
For both of these tests, I was able to reproduce the failures consistently on my local box by inserting a long sleep between the `(deliver lock-thread-started? true)` call and the `(.lock pool)` call.  I used a 9 second sleep - long enough to reliably cause the other `borrow-thread-*` futures to be able to make a `.borrowItem` calls - and for one to succeed after the `.releaseItem` call for the initially borrowed instance was made.  Less than 10 seconds was important to use in order to avoid hitting the "global" `timed-deref` timeout of 10 seconds that these tests arbitrarily use.